doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+
+*  Alarms
+
+   Alarms are notifications raised for conditions that could be cleared by correcting or removal of such conditions.
+   
+   Out of memory, temperature crossing a threshold, and so on, are examples of conditions when the alarms are raised.
+   Such conditions are dynamic: a faulty software/hardware component encounters the above such condition and **may** come out of that situation when the condition is resolved.
+   
+   Events are sent as the condition progresses through being raised and cleared in addition to operator acknowledging/unacknowledging it.
+   So, these events have a field called *action*: RAISE, CLEAR or ACKNOWLEDGE/UNACKNOWLEDGE.
+
+   Each of such events for an alarm is characterized by "action" in addition to "severity".
+   
+   An application *raises* an alarm when it encounters a faulty condition by sending an event with action: *RAISE*.
+   After the application recovers from the condition, that alarm is *cleared* by sending an event with action: *CLEAR*.
+   An operator could *ACKNOWLEDGE/UNACKNOWLEDGE* an alarm. This indicates that the operator is aware of the faulty condition.
+
+   The set of alarms and their severities are an indication to health of various applications of the system and System LED can be deduced from alarms.
+   An acknowledged alarm means that operator is aware of the condition so, acknowledged alarm will be taken out of consideration.
+
+Both events and alarms get recorded in a new DB called EVENT DB in a new redis instance.
+
+1.  Event History Table
+
+    All events get recorded in the event history table, by name, "EVENT". EVENT table contains history of all events generated by the system.
+    This table is persisted across system restarts of any kind, including restore to factory defaults and SW upgrades and downgrades.
+
+2. Current Alarm Table
+
+   All events with an action field of *RAISE* get recorded in a table, by name, "ALARM" in addition to getting recorded in Event History Table ( only events corresponding to an alarm has action field ).
+   When an application that raised the alarm clears it ( by sending an event with action *CLEAR* ), the alarm record is removed from ALARM table.
+   An user acknowledging a particular alarm will NOT remove that alarm record from this table; only when application clears it, the alarm is removed from ALARM table.
+   
+   In effect, ALARM table contains outstanding alarms that need to be cleared by those applications who raised them.
+   This table is NOT persisted and its contents are cleared with a reload.
+
+In summary, the framework provides both current and historical event status of software and physical entities of the system through ALARM and EVENT tables. 
+
+In addition to the above tables, the framework maintains various statisitcs.
+
+1. Event Statistics Table
+
+   Statistics on number of events and alarms are maintained in EVENT_STATS table.
```

Comment:
`Event` table containes the statistics of `alarms`?

Reply:
Updated.